src/afo-register/domain/Record.ts

```diff
@@ -0,0 +1,94 @@
+import produce, { Draft, immerable } from 'immer'
+
+interface RecordData {
+  readonly afoNumber: string
+  readonly page: string
+  readonly text: string
+  readonly textNumber: string
+  readonly linesDiscussed?: string
+  readonly discussedBy?: string
+  readonly discussedByNotes?: string
+  readonly fragmentNumbers?: string[]
+}
+
+export class AfoRegisterRecordSuggestion {
+  [immerable] = true
+
+  readonly text: string
+  readonly textNumbers: string[]
+
+  constructor({
+    text,
+    textNumbers,
+  }: {
+    readonly text: string
+    readonly textNumbers: string[]
+  }) {
+    this.text = text
+    this.textNumbers = textNumbers
+  }
+}
+
+export default class AfoRegisterRecord {
+  [immerable] = true
+
+  readonly afoNumber: string
+  readonly page: string
+  readonly text: string
+  readonly textNumber: string
+  readonly linesDiscussed?: string
+  readonly discussedBy?: string
+  readonly discussedByNotes?: string
+  readonly fragmentNumbers?: string[]
+
+  constructor({
+    afoNumber,
+    page,
+    text,
+    textNumber,
+    linesDiscussed,
+    discussedBy,
+    discussedByNotes,
+    fragmentNumbers,
+  }: RecordData) {
+    this.afoNumber = afoNumber
+    this.page = page
+    this.text = text
+    this.textNumber = textNumber
+    this.linesDiscussed = linesDiscussed
+    this.discussedBy = discussedBy
+    this.discussedByNotes = discussedByNotes
+    this.fragmentNumbers = fragmentNumbers
+  }
+
+  toMarkdownString(): string {
+    let result = this.text + (this.textNumber ? ' ' + this.textNumber : '')
```

Comment:
I recently made a little helper function `padLeft` in src/fragmentarium/domain/archaeology.ts because I had to do the `x ? ' ' + x : ''` thing over and over, which is then just `padLeft(this.textNumber)`. If you like you can move it to common/utils.ts or something and use it here, it might lower the complexity a bit.